v1/pipeline.json

```diff
@@ -50682,6 +50691,265 @@
             "type" : "string"
           }
         }
+      },
+      "Notification Rules" : {
+        "title" : "Notification Rules",
+        "type" : "object",
+        "required" : [ "method", "enabled" ],
+        "properties" : {
+          "pipeline_events" : {
+            "type" : "array",
+            "items" : {
+              "type" : "string",
+              "enum" : [ "start", "end", "success", "failed", "all" ]
+            }
+          },
+          "stage_events" : {
+            "type" : "array",
+            "items" : {
+              "$ref" : "#/definitions/pipeline/Stage Event Notification"
+            }
+          },
+          "step_events" : {
+            "type" : "string",
+            "enum" : [ "failed" ]
+          },
+          "method" : {
+            "$ref" : "#/definitions/pipeline/NotificationChannelWrapper"
+          },
+          "enabled" : {
+            "type" : "boolean"
+          }
+        }
+      },
+      "Stage Event Notification" : {
+        "title" : "Stage Event Notification",
+        "type" : "object",
+        "required" : [ "type" ],
+        "properties" : {
+          "type" : {
+            "type" : "string",
+            "enum" : [ "start", "success", "failed", "all" ],
+            "refs" : {
```

Comment:
This `refs` looks confusing like what refs are these. Shall we call this `stageIds` or `stageRefs` or `stages`?

Reply:
since this is under the key `stage_events`, with just `refs` it is implied that these are `stage_refs`.
But if we think it is confusing, we can change it to `stages`. 
@archit-harness wdyt?

Reply:
we are following a practice to not explicitly mention it, rather @shalini-agr you should add description field mentioning which ref this is.

cc: @SudarshanPurohit

Reply:
done